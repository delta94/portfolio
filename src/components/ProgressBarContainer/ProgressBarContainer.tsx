import React from 'react';
import { createUseStyles } from 'react-jss';
import { CustomTheme } from '../../theme';

interface ProgressBarContainerProps {
  name?: string;
  progress?: number;
  backgroundColor?: string;
}

const useStyles = createUseStyles((theme: CustomTheme) => ({
  mainContainer: {
    width: '100%',
    paddingLeft: theme.spacing(2.5),
  },
  description: {
    display: 'flex',
    justifyContent: 'space-between',
    width: '100%',
  },
  progressBarContainer: {
    backgroundColor: theme.colors.white,
    height: 8,
    borderRadius: 20,
  },
  progressBar: {
    backgroundColor: ({ backgroundColor }: ProgressBarContainerProps): string =>
      backgroundColor ? backgroundColor : theme.colors.white,
    height: '100%',
    borderRadius: 20,
    width: ({ progress }: ProgressBarContainerProps): string => (progress ? `${progress}%` : '100%'),
  },
  progressBarParent: {
    width: '100%',
  },
}));
const UnMemoizedProgressBarContainer: React.FC<ProgressBarContainerProps> = (props) => {
  const classes: Record<
    'mainContainer' | 'description' | 'progressBarContainer' | 'progressBar' | 'progressBarParent',
    string
  > = useStyles(props);

  const { name, progress } = props;

  return (
    <div className={classes.progressBarParent}>
      <div className={classes.description}>
        <h3>{name}</h3>
        <h3>{progress}%</h3>
      </div>
      <div className={classes.progressBarContainer}>
        <div className={classes.progressBar} />
      </div>
    </div>
  );
};

const ProgressBarContainer: React.NamedExoticComponent<ProgressBarContainerProps> = React.memo(
  UnMemoizedProgressBarContainer,
);

export { ProgressBarContainer };
