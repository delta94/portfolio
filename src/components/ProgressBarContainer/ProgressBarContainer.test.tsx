import { render } from '@testing-library/react';
import React from 'react';
import { TestContextProvider } from '../../TestContextProvider';
import { ProgressBarContainer } from './ProgressBarContainer';

const mockData = {
  name: 'mock name',
  backgroundColor: 'blue',
  progress: 50,
};
test('ProgressBarContainer renders correctly', () => {
  const { asFragment } = render(
    <TestContextProvider>
      <ProgressBarContainer {...mockData} />
    </TestContextProvider>,
  );
  expect(asFragment()).toMatchSnapshot();
});
