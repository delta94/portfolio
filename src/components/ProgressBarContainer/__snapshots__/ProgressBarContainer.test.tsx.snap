// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProgressBarContainer renders correctly 1`] = `
<DocumentFragment>
  <div
    class="progressBarParent-0-2-5"
  >
    <div
      class="description-0-2-2"
    >
      <h3>
        mock name
      </h3>
      <h3>
        50%
      </h3>
    </div>
    <div
      class="progressBarContainer-0-2-3"
    >
      <div
        class="progressBar-0-2-4 progressBar-d0-0-2-6"
      />
    </div>
  </div>
</DocumentFragment>
`;
